src/common.jl

```diff
@@ -1,80 +1,126 @@
-struct LinearCache{TA,Tb,Tp,Talg,Tc,Tr,Tl}
+struct LinearCache{TA,Tb,Tu,Tp,Talg,Tc,Tl,Tr}
     A::TA
     b::Tb
+    u::Tu
     p::Tp
     alg::Talg
-    cacheval::Tc
-    isfresh::Bool
-    Pr::Tr
-    Pl::Tl
+    cacheval::Tc  # store alg cache here 
+    isfresh::Bool # false => cacheval is set wrt A, true => update cacheval wrt A
+#
+    Pl::Tl        # store final preconditioner here. not being used rn
+    Pr::Tr        # wrappers are using preconditioner in cache.alg for now
 end
 
-function set_A(cache, A)
+function set_A(cache::LinearCache, A) # and ! to function name
```

Comment:
```suggestion
function set_A(cache::LinearCache, A)
```
We don't add `!` to non-mutating function.